Makefile

```diff
@@ -80,18 +78,18 @@ FISHINSTALLDIR=${PREFIX}/share/fish/vendor_completions.d
 SELINUXOPT ?= $(shell test -x /usr/sbin/selinuxenabled && selinuxenabled && echo -Z)
 
 COMMIT_NO ?= $(shell git rev-parse HEAD 2> /dev/null || true)
-GIT_COMMIT ?= $(if $(shell git status --porcelain --untracked-files=no),${COMMIT_NO}-dirty,${COMMIT_NO})
+GIT_COMMIT ?= $(if $(shell git status --porcelain --untracked-files=no),$(call err_if_empty,COMMIT_NO)-dirty,$(COMMIT_NO))
 DATE_FMT = %s
 ifdef SOURCE_DATE_EPOCH
-	BUILD_INFO ?= $(shell date -u -d "@$(SOURCE_DATE_EPOCH)" "+$(DATE_FMT)" 2>/dev/null || date -u -r "$(SOURCE_DATE_EPOCH)" "+$(DATE_FMT)" 2>/dev/null || date -u "+$(DATE_FMT)")
+	BUILD_INFO ?= $(shell date -u -d "@$(call err_if_empty,SOURCE_DATE_EPOCH)" "+$(DATE_FMT)" 2>/dev/null || date -u -r "$(SOURCE_DATE_EPOCH)" "+$(DATE_FMT)" 2>/dev/null || date -u "+$(DATE_FMT)")
 else
 	BUILD_INFO ?= $(shell date "+$(DATE_FMT)")
 endif
 LIBPOD := ${PROJECT}/v4/libpod
 GOFLAGS ?= -trimpath
 LDFLAGS_PODMAN ?= \
-	-X $(LIBPOD)/define.gitCommit=$(GIT_COMMIT) \
-	-X $(LIBPOD)/define.buildInfo=$(BUILD_INFO) \
+	-X $(LIBPOD)/define.gitCommit=$(call err_if_empty,GIT_COMMIT) \
+	-X $(LIBPOD)/define.buildInfo=$(call err_if_empty,BUILD_INFO) \
```

Comment:
I don't think we can force this, we should never fail if there is not git information. This is just extra but not required. 
https://github.com/containers/podman/issues/14021#issuecomment-1110698996

Reply:
And why are you changing `gitCommit` to `buildInfo`? The log colorizer depends on `gitCommit`

Reply:
hhu?  No change intended, just checking if it's empty string or not.  Something I'm not seeing?

> I don't think we can force this

I'm okay with that, but it does add some build ambiguity since "sometimes" we do want to require it, for example when building packages/binaries for release.  Maybe that could be addressed by adding a release-test to check for it.

Reply:
My bad: I misread the diff. Sorry. Agree with @Luap99 that this cannot be an error if empty. Adding a test later is fine; it would be a trivial addition to https://github.com/containers/podman/blob/ccc087a30e7f84d656663e0b60cd2b918f5288fd/test/system/001-basic.bats#L28